Arduino IDE Add-on : Serial Monitor for ESP8266

For the ones interested in I've recently developed an add-on for the ARDUINO IDE in order to be able to use the Serial Monitor with an ESP8266 generic module.

![menu](https://cloud.githubusercontent.com/assets/13887597/12079820/9b2a0784-b247-11e5-8984-33c422629586.png)
The add-on is available in the menu under 'Tools> ESP8266 Modify Serial Monitor' and starts a modified version of the Serial Monitor with 3 extra widgets:
- A 'clear' button to reset the content of the terminal
- A 'reset' check-box to set the status of the RTS signal (RESET)
- A 'programmation' check-box to set the status of the DTR signal

NOTE : Arduino Serial Monitor shall not be started otherwise modified one will not be available. Restart the IDE if you started the Arduino Serial Monitor before.

On start 'reset' is checked and 'programmation' is none that will reset the ESP8266 . Un-check 'reset' to get the board running.

![modified serial monitor](https://cloud.githubusercontent.com/assets/13887597/12079821/9b2e5cf8-b247-11e5-84a2-3ad2d40fbf7b.png)

If the community agrees and help me (I've never committed on GITHUB) I can publish the source code for the add-on.

Extract the following archive in the tools folder of your ARDUINO sketch folder.
http://mytrain.fr/cms//images/mytrain/private/ESP8266SM.zip
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29554852-arduino-ide-add-on-serial-monitor-for-esp8266?utm_campaign=plugin&utm_content=tracker%2F14245935&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F14245935&utm_medium=issues&utm_source=github).
</bountysource-plugin>

Ooops!
````
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: processing/app/debug/TextAreaFIFO
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.getDeclaredMethod(Class.java:2128)
	at java.awt.Component.isCoalesceEventsOverriden(Component.java:6220)
	at java.awt.Component.access$500(Component.java:186)
	at java.awt.Component$3.run(Component.java:6174)
	at java.awt.Component$3.run(Component.java:6172)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.Component.checkCoalescing(Component.java:6171)
	at java.awt.Component.<init>(Component.java:6140)
	at java.awt.Container.<init>(Container.java:283)
	at java.awt.Window.<init>(Window.java:535)
	at java.awt.Frame.<init>(Frame.java:420)
	at javax.swing.JFrame.<init>(JFrame.java:233)
	at processing.app.AbstractMonitor.<init>(AbstractMonitor.java:26)
	at processing.app.AbstractTextMonitor.<init>(AbstractTextMonitor.java:40)
	at processing.app.SerialMonitor.<init>(SerialMonitor.java:37)
	at org.d2002.esp8266.arduino.serialmonitor.ESP8266SerialMonitor.<init>(ESP8266SerialMonitor.java:39)
	at org.d2002.esp8266.arduino.serialmonitor.ESP8266SM.createMonitor(ESP8266SM.java:120)
	at org.d2002.esp8266.arduino.serialmonitor.ESP8266SM.run(ESP8266SM.java:108)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.ClassNotFoundException: processing.app.debug.TextAreaFIFO
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 34 more
````

--- a/README.md
+++ b/README.md
@@ -123,3 +123,13 @@ ESP8266 core files are licensed under LGPL.
 [BearSSL](https://bearssl.org) library written by Thomas Pornin, built from https://github.com/earlephilhower/bearssl-esp8266, is used in this project.  It is distributed under the [MIT License](https://bearssl.org/#legal-details).
 
 [LittleFS](https://github.com/ARMmbed/littlefs) library written by ARM Limited and released under the [BSD 3-clause license](https://github.com/ARMmbed/littlefs/blob/master/LICENSE.md).
+
+### Other useful links ###
+
+[Toolchain repo](https://github.com/earlephilhower/esp-quick-toolchain)
+
+[Lwip link layer repo](https://github.com/d-a-v/esp82xx-nonos-linklayer)
+
+[SoftwareSerial repo](https://github.com/plerup/espsoftwareserial)
+
+[Serial Monitor Arduino IDE plugin](https://github.com/mytrain/arduino-esp8266-serial-plugin) Original discussion [here](https://github.com/esp8266/Arduino/issues/1360), quick download [there](http://mytrain.fr/cms//images/mytrain/private/ESP8266SM.v3.zip).

